manager/volume.go

```diff
@@ -562,7 +561,14 @@ func (m *VolumeManager) Expand(volumeName string, size int64) (v *longhorn.Volum
 		return v, nil
 	}
 
-	if err := m.scheduler.CheckReplicasSizeExpansion(v, v.Spec.Size, size); err != nil {
+	multiError, err := m.scheduler.CheckReplicasSizeExpansion(v, v.Spec.Size, size)
+	if err != nil {
+		if multiError != nil {
+			failureMessage := multiError.Join()
+			if err := m.ds.UpdatePVAnnotation(v, types.PVAnnotationLonghornVolumeSchedulingError, failureMessage); err != nil {
+				logrus.Warnf("Cannot update PV annotation for volume %v", volumeName)
+			}
+		}
```

Comment:
I don't think we need to update during the validation phase, but rather only do this during the volume size reconciling, because a validation failure just means the request has not been accepted and nothing changes to the volume resource change, right?

Reply:
I think yes. The volume is at detached state, and PV/PVC STATUS are `Bound` in the validation failure case. So, the Havester should check the status of the volume (I mean detached`), right?

Reply:
Should Harvester check the PVC error message (kubectl describe pvc ...) rather than volume-scheduling-error label?

Reply:
If the PCV is bound but the corresponding volume replicas can't be scheduled, the downstream users will be unable to check any error messages from the PVC. 

Therefore, the only way is to rely on checking the volume status condition `scheduled` or the updated annotation ` volume-scheduling-error` of the PVC. 

Anything missing?

Reply:
Yeah, if we don't set `volume-scheduling-error`, checking volume status is the only way.

Reply:
Updated. Thank you.